mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -918,9 +924,31 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
     }
 
+    private void restartAfterSettingsUpdate() {
+        restartAfterSettingsUpdate(false);
+    }
+
+    private void restartAfterSettingsUpdate(boolean firstTimeDemo) {
+        // Restart app after preferences
+        Log.d(TAG, "Restarting after settings");
+        // Get launch intent for application
+        Intent restartIntent = getPackageManager().getLaunchIntentForPackage(getPackageName());
+        restartIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        if (firstTimeDemo) {
+            restartIntent.putExtra(EXTRA_DEMO_FIRST_TIME, true);
+        }
+        // Finish current activity
+        finish();
+        // Start launch activity
+        startActivity(restartIntent);
```

Comment:
Just use recreate() if firstTimeDemo is false.

Reply:
Done!